warehouse/warehouse.go

```diff
@@ -1233,16 +1232,33 @@ func pendingEventsHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	filterBy := []warehouseutils.FilterBy{{Key: "source_id", Value: sourceID}}
-	if pendingEventsReq.TaskRunID != "" {
-		filterBy = append(filterBy, warehouseutils.FilterBy{Key: "metadata->>'source_task_run_id'", Value: pendingEventsReq.TaskRunID})
+	filters := []repo.FilterBy{
+		{Key: "source_id", Value: sourceID},
```

Comment:
Have we defined the table columns as constants? Maybe it would be safer to use constants.